internal/sample/sample.go

```diff
@@ -0,0 +1,115 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sample
+
+import (
+	"fmt"
+	"github.com/GoogleCloudPlatform/serverless-sample-tester/internal/gcloud"
+	"github.com/GoogleCloudPlatform/serverless-sample-tester/internal/lifecycle"
+	"github.com/GoogleCloudPlatform/serverless-sample-tester/internal/util"
+	"os/exec"
+	"strings"
+	"unicode"
+)
+
+//
+const maxCloudContainerImageTagLen = 53
+
+// Sample represents a Google Cloud Platform sample and associated properties.
+type Sample struct {
```

Comment:
Just wanted to say that I'm happy with how you simplified the surface here-- this looks much better than what you had before.

Also, moving these types to an internal directory also was a good move probably.